packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -0,0 +1,285 @@
+  const queryArguments = {
+    query: raw('$util.toJson($query)'),
+    scanIndexForward: ifElse(
+      ref('context.args.sortDirection'),
+      ifElse(equals(ref('context.args.sortDirection'), str('ASC')), bool(true), bool(false)),
+      bool(true),
+    ),
+    filter: ifElse(ref('context.args.filter'), ref('util.transform.toDynamoDBFilterExpression($ctx.args.filter)'), nul()),
+    limit: ref('limit'),
+    nextToken: ifElse(ref('context.args.nextToken'), ref('util.toJson($context.args.nextToken)'), nul()),
+  } as any;
+
+  if (indexName) {
+    queryArguments.index = str(indexName);
+  }
+
+  const queryObj = DynamoDBMappingTemplate.query(queryArguments);
+  const resolver = ctx.resolvers.generateQueryResolver(
+    object.name.value,
+    field.name.value,
+    dataSource as any,
+    MappingTemplate.s3MappingTemplateFromString(
+      print(
+        ifElse(
+          raw(`$util.isNull($ctx.source.${connectionAttributes[0]})`),
+          compoundExpression([set(ref('result'), obj({ items: list([]) })), raw('#return($result)')]),
+          compoundExpression([...setup, queryObj]),
+        ),
+      ),
+      `${object.name.value}.${field.name.value}.req.vtl`,
+    ),
+    MappingTemplate.s3MappingTemplateFromString(
+      print(
+        DynamoDBMappingTemplate.dynamoDBResponse(
+          false,
+          compoundExpression([iff(raw('!$result'), set(ref('result'), ref('ctx.result'))), raw('$util.toJson($result)')]),
+        ),
+      ),
+      `${object.name.value}.${field.name.value}.res.vtl`,
+    ),
+  );
+
+  resolver.mapToStack(table.stack);
+  ctx.resolvers.addResolver(object.name.value, field.name.value, resolver);
+}
+
+function makeExpression(keySchema: any[], connectionAttributes: string[]): ObjectNode {
+  if (keySchema[1] && connectionAttributes[1]) {
+    let condensedSortKeyValue;
+
+    if (connectionAttributes.length > 2) {
+      const rangeKeyFields = connectionAttributes.slice(1);
+
+      condensedSortKeyValue = rangeKeyFields
+        .map(keyField => `\${context.source.${keyField}}`)
+        .join(ModelResourceIDs.ModelCompositeKeySeparator());
+    }
+
+    return obj({
+      expression: str('#partitionKey = :partitionKey AND #sortKey = :sortKey'),
+      expressionNames: obj({
+        '#partitionKey': str(keySchema[0].attributeName),
+        '#sortKey': str(keySchema[1].attributeName),
+      }),
+      expressionValues: obj({
+        ':partitionKey': obj({
+          S: str(`$context.source.${connectionAttributes[0]}`),
+        }),
+        ':sortKey': obj({
+          S: str(condensedSortKeyValue || `$context.source.${connectionAttributes[1]}`),
+        }),
+      }),
+    });
+  }
+
+  return obj({
+    expression: str('#partitionKey = :partitionKey'),
+    expressionNames: obj({
+      '#partitionKey': str(keySchema[0].attributeName),
+    }),
+    expressionValues: obj({
+      ':partitionKey': obj({
+        S: str(`$context.source.${connectionAttributes[0]}`),
+      }),
+    }),
+  });
+}
```

Comment:
Nit:  Can we use `$util.dynamodb.toDynamoDB(Object)` here for the expression values?